common/tedge_users/src/unix.rs

```diff
@@ -59,6 +59,22 @@ impl UserManager {
         users::get_current_uid() == 0
     }
 
+    /// Check if the process has been launched using `tedge-mapper` or not.
+    ///
+    /// # Example
+    ///
+    /// ```
+    ///     # use tedge_users::UserManager;
+    ///     let path = if UserManager::running_as_tedge_mapper() {
+    ///          "/etc/tedge/tedge.toml"
+    ///      } else {
+    ///          ".tedge/tedge.toml"
+    ///      };
+    /// ```
+    pub fn running_as_tedge_mapper() -> bool {
```

Comment:
I'd suggest that you take the `desired_user` as a parameter of this API and compare if the `current_user` is the `desired_user`. That way a `collectd_mapper` process can check if the current user is `collect-mapper` and the cloud mappers can check if the current user is `tedge-mapper`.